config/aso/credentials.yaml

```diff
@@ -4,7 +4,7 @@ metadata:
   name: aso-controller-settings
 type: Opaque
 data:
-  AZURE_SUBSCRIPTION_ID: ${AZURE_SUBSCRIPTION_ID_B64:=""}
-  AZURE_TENANT_ID: ${AZURE_TENANT_ID_B64:=""}
-  AZURE_CLIENT_ID: ${AZURE_CLIENT_ID_B64:=""}
-  AZURE_CLIENT_SECRET: ${AZURE_CLIENT_SECRET_B64:=""}
+  # Per-resource Secrets will be created based on a Cluster's AzureClusterIdentity.
+  AZURE_SUBSCRIPTION_ID: ""
```

Comment:
uh is that behavior we expect? if not is there an issue tracking this in ASO to make them optional?

Reply:
I don't remember testing this very thoroughly since https://github.com/Azure/azure-service-operator/issues/2725 was fixed, but it looks like we can probably omit this Secret entirely.

Reply:
Actually I see this error when the Secret doesn't exist at all, so I think we need to keep this with blank values. I'm following up in that PR fixing that issue: https://github.com/Azure/azure-service-operator/pull/2758#issuecomment-1691944668. That seems like a bug but I'll keep this as-is for now until we have a fix.

Reply:
> this error

An event actually:
```
89s         Warning   Failed                  pod/azureserviceoperator-controller-manager-84446f4877-z8jw7    Error: secret "aso-controller-settings" not found
```